src/handle.js

```diff
@@ -1,6 +1,25 @@
 const handle = (help) => {
-    // TODO: create implementation of parse function
+    const parseSection = require('./parseSection.js');
+    const parseOption = require('./parseOption');
+    const sectionName = "Options:";
+
     let options = [];
+    parseSection(sectionName, help).forEach(section => {
+        let splitTmp = (section.split(/\n[ \t]*(-\S+?)/)).slice(1);
+        let split = [];
+        for (let cnt = splitTmp.length, i = 0; i < cnt; i+=2) {
+            if (splitTmp !== undefined) {
+                split.push(splitTmp[i] + splitTmp[i + 1]);
```

Comment:
Probably `[ splitTmp[i] , splitTmp[i + 1] ]`?

Reply:
I'm trying to follow the logic here, split then concat back, what is the idea?

Reply:
Sorry, i forgot about set flag PREG_SPLIT_DELIM_CAPTURE of preg_split php function when transating it to js. If this flag is set, parenthesized expression in the delimiter pattern will be captured and returned as well. This is also an answer on your question about parenthes in regular expression.

For example, with reg exp /\n[ \t]*(-\S+?)/ string
"-c, --config          configuration file                                                                                
  -e, --exclude         exclude globs from path expansion"                                                              
will be sliced into 4 strings:  
* "-c" 
* ", --config          configuration file"
* "-e" 
* ", --exclude         exclude globs from path expansion"                                                           

So, its needed to concate 1 and 2, 3 and 4 strings later

Maybe you know, what is js alternative for PREG_SPLIT_DELIM_CAPTURE flag or can suggest another solution of this?

Reply:
Like this one?
https://searchcode.com/codesearch/view/14390265/#

There is no alternative out of the box.